### ISSUE TITLE
Unhelpful behavior when including multiple BAs

### ISSUE THREAD

**Issue by rseanhall** [wixtoolset/issues#6305]
If this issue is a bug:

* Which version of WiX are you building with?

> WixToolset.Sdk.4.0.0-build-0175

* Which version of Visual Studio are you building with (if any)?

> N/A

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> N/A

* Which version of .NET are you building with?

> N/A

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> N/A

* Describe the problem and the steps to reproduce it.

> Include multiple `BootstrapperApplicationDll` elements in a bundle.

* Describe the behavior you expected and how it differed from the actual behavior.

According to the comment, this should be detected at link time:

```
// Ensure there is one and only one row in the WixBootstrapperApplicationDll table.
// The compiler and linker behavior should have colluded to get this behavior.
```

Instead, this unhelpful message is shown at bind time which is also misleading because the problem is too much data not missing data:

> error WIX341: The Bundle is missing 'T' data, and cannot continue.

**Comment by barnson** [comment]
What?! It's perfectly accurate and precise, for certain values of `T`.

**Comment by rseanhall** [comment]
I couldn't figure out to make this happen in a test in Core, there must be something special with the Bal.wixext BA's. This is the only way I know how to reproduce - https://github.com/rseanhall/burn-e2e-tests/tree/issue-6305.

**Comment by barnson** [comment]
I don't think this is possible any more. `ParseBootstrapperApplicationDllElement` always creates a `WixBundleContainerSymbol` so this now creates a duplicate symbol error.

**Comment by rseanhall** [comment]
I still don't see a duplicate symbol error. I have created a new repro branch at https://github.com/rseanhall/wix4/tree/issue-6305-2. It fails differently now but I think it's still after duplicate symbol errors should have stopped the build.

At https://github.com/rseanhall/wix4/blob/0e18592839a2ef05527caf4658e6fa5d3d1cb6ab/src/wix/WixToolset.Core/Link/FlattenAndProcessBundleTablesCommand.cs#L64, `ArgumentException: An item with the same key has already been added.`
